delta_backend/src/extractor.py

```diff
@@ -0,0 +1,409 @@
+import json
+import exception_messages
+from datetime import datetime, timedelta, timezone
+from common.mappings import Gender
+
+class Extractor: 
+
+    # This file holds the schema/base layout that maps FHIR fields to flat JSON fields
+    # Each entry tells the converter how to extract and transform a specific value
+    EXTENSION_URL_VACCINATION_PRODEDURE = "https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-VaccinationProcedure"
+    EXTENSION_URL_SCT_DESC_DISPLAY = "https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-CodingSCTDescDisplay"
+
+    CODING_SYSTEM_URL_SNOMED = "http://snomed.info/sct"
+    ODS_ORG_CODE_SYSTEM_URL = "https://fhir.nhs.uk/Id/ods-organization-code"
+    DEFAULT_LOCATION = "X99999"
+    NHS_NUMBER_SYSTEM_URL = "https://fhir.nhs.uk/Id/nhs-number"
+    
+    DATE_CONVERT_FORMAT = "%Y%m%d"
+    DEFAULT_POSTCODE = "ZZ99 3CZ"
+    
+    def __init__(self, fhir_json_data, report_unexpected_exception = True):
+        self.fhir_json_data = json.loads(fhir_json_data) if isinstance(fhir_json_data, str) else fhir_json_data
+        self.report_unexpected_exception = report_unexpected_exception
+        self.error_records = []
+        
+    def _get_patient(self):
+        contained = self.fhir_json_data.get("contained", [])
+        return next((c for c in contained if isinstance(c, dict) and c.get("resourceType") == "Patient"), "")
+
+    def _get_valid_names(self, names, occurrence_time):  
```

Comment:
The spec has one extra rule - only consider names which have a `given` and `family`. I feel like we should question this though - follow up ticket?